src/DependencyInjection/HttplugExtension.php

```diff
@@ -506,13 +531,79 @@ private function configurePlugin(ContainerBuilder $container, $serviceId, $plugi
     {
         $pluginServiceId = $serviceId.'.plugin.'.$pluginName;
 
-        $definition = class_exists(ChildDefinition::class)
-            ? new ChildDefinition('httplug.plugin.'.$pluginName)
-            : new DefinitionDecorator('httplug.plugin.'.$pluginName);
+        $definition = $this->getChildDefinition('httplug.plugin.'.$pluginName);
 
         $this->configurePluginByName($pluginName, $definition, $pluginConfig, $container, $pluginServiceId);
         $container->setDefinition($pluginServiceId, $definition);
 
         return $pluginServiceId;
     }
+
+    private function configureVcrPlugin(ContainerBuilder $container, array $config, $prefix)
+    {
+        $recorder = $config['recorder'];
+        $recorderId = $container->hasDefinition($recorder) ? $recorder : 'httplug.plugin.vcr.recorder.'.$recorder;
```

Comment:
i'd say `in_array($recorder, ['filesystem', 'in_memory']) ? 'httplug.plugin.vcr.recorder.'.$recorder : $recorder;`

otherwise we break on somebody define filesystem / in_memory as a service in their application, and we end up with a confusing error if the service is not defined. also, hasDefinition depends on the order in which configuration is loaded, the $recorder service could be defined later than this bundle is loaded.